src/table/internal.tsx

```diff
@@ -149,6 +159,63 @@ const InternalTable = React.forwardRef(
     const hasSelection = !!selectionType;
     const hasFooter = !!footer;
 
+    // Allows keyboard users to scroll horizontally with arrow keys by making the wrapper part of the tab sequence
+    const isWrapperScrollable = !!tableWidth && !!containerWidth && tableWidth > containerWidth;
+    const wrapperProps = isWrapperScrollable
+      ? { role: 'region', tabIndex: 0, 'aria-label': ariaLabels?.tableLabel }
+      : {};
+
+    const [tablePadding, setTablePadding] = useState({ left: 0, right: 0 });
+    useEffect(() => {
+      if (!tableRefObject.current) {
+        return;
+      }
+      const left = parseInt(getComputedStyle(tableRefObject.current).paddingLeft) || 0;
+      const right = parseInt(getComputedStyle(tableRefObject.current).paddingRight) || 0;
+      setTablePadding({ left, right });
+    }, [tableRefObject]);
+
+    const getMouseDownTarget = useMouseDownTarget();
+    const wrapWithInlineLoadingState = (submitEdit: TableProps['submitEdit']) => {
+      if (!submitEdit) {
+        return undefined;
+      }
+      return async (...args: Parameters<typeof submitEdit>) => {
+        setCurrentEditLoading(true);
+        try {
+          await submitEdit(...args);
+        } finally {
+          setCurrentEditLoading(false);
+        }
+      };
+    };
+
+    const hasDynamicHeight = computedVariant === 'full-page';
+    const overlapElement = useDynamicOverlap({ disabled: !hasDynamicHeight });
+
+    const stickyColumnsParams = {
+      visibleColumns,
+      visibleColumnsLength,
+      hasSelection,
+      stickyColumns,
+      tableRefObject,
+      containerWidth,
+      isWrapperScrollable,
+      tablePadding,
+      wrapperRef: wrapperRefObject,
+    };
+
+    const {
+      tableCellRefs,
+      getStickyColumnProperties,
+      shouldDisableStickyColumns,
+      wrapperScrollPadding,
+      updateCellOffsets,
+      cellOffsets,
+    } = useStickyColumns(stickyColumnsParams);
```

Comment:
Regarding performance, 

I added simple debugging:
```
useEffect(() => {
  console.log('id');
}, [baseProps.id]);
useEffect(() => {
  console.log('tableCellRefs');
}, [tableCellRefs]);
useEffect(() => {
  console.log('getStickyColumnProperties');
}, [getStickyColumnProperties]);
useEffect(() => {
  console.log('shouldDisableStickyColumns');
}, [shouldDisableStickyColumns]);
useEffect(() => {
  console.log('wrapperScrollPadding');
}, [wrapperScrollPadding]);
useEffect(() => {
  console.log('updateCellOffsets');
}, [updateCellOffsets]);
useEffect(() => {
  console.log('cellOffsets');
}, [cellOffsets]);
```

When rendering a table w/o sticky columns, it outputs:
```
id
tableCellRefs
getStickyColumnProperties
shouldDisableStickyColumns
wrapperScrollPadding
updateCellOffsets
cellOffsets
tableCellRefs // x2
getStickyColumnProperties // x2
updateCellOffsets // x2
```

When rendering a table w/ sticky columns, it outputs:
```
id
tableCellRefs
getStickyColumnProperties
shouldDisableStickyColumns
wrapperScrollPadding
updateCellOffsets
cellOffsets
tableCellRefs // x2
getStickyColumnProperties // x2
shouldDisableStickyColumns // x2
updateCellOffsets // x2
cellOffsets // x2
getStickyColumnProperties // x3
shouldDisableStickyColumns // x3
getStickyColumnProperties // x4
wrapperScrollPadding // x2
cellOffsets // x3
getStickyColumnProperties // x4
cellOffsets // x4
```

That contributes to additional rendering of the dependant elements. Ideally, all properties needs to be rendered exactly once. Practically, because you need the table to render before certain calculations are possible, some of the properties will unavoidable render twice, but only when the feature is configured.

Reply:
One possible way to solve it would be to use Redux-like subscription mechanism not only for table cells but also for other elements to where the sticky-columns state is dispatched.

For example:

```
function Table(...) {
  // ...
  // The sticky state store is created one and is never updates. However, its internal state is updated reactively.
  const stickyStateStore = useStickyState(stickyColumnsParams)
  // ...
 return (<ColumnWidthsProvider ...>
   <InternalContainer ...>
     <TableWrapper stickyState={stickyStateStore} ...>
       <table ...>
         <Thead .../>
         <tbody>
           ...
             <TableTdElement stickyState={stickyStateStore} ... />
         </tbody>
       </table>
     </TableWrapper>
   </InternalContainer>
 </ColumnWidthsProvider>)
}

function TableWrapper({ children, stickyState }) {
  useEffect(() => {
    stickyState.subscribe('scrollPaddingLeft', scrollPaddingLeft => ...)
    stickyState.subscribe('scrollPaddingRight', scrollPaddingLeft => ...)
    return () => unsubscribe(...)
  }, [])
  
  return <div ...>{children}</div>
}

function TableTdElement({ stickyState, ... }) {
  useEffect(() => {
    stickyState.subscribe('stickyCellClasses', stickyCellClasses => ...)
    stickyState.subscribe('stickyCellOffset', stickyCellOffset => ...)
    return () => unsubscribe(...)
  }, [])
  
  return <td ...>...</td>
}
```

Reply:
In that case the sticky state store can encapsulate the whole logic. You can pass it directly or distribute via React context. If distribute directly there is no need in having sticky-context, use-sticky-columns and use-sticky-state - just one would be enough.

This approach is used in the area-chart, the store is represented with the ChartModel interface: https://github.com/cloudscape-design/components/blob/main/src/area-chart/model/index.ts#L11 and here is an example of a subscription: https://github.com/cloudscape-design/components/blob/main/src/area-chart/elements/vertical-marker.tsx#L14

That is the opposite of updating the state in the root of the component and then passing it down.